test/Middleware/AuthHttpBasicTest.php

```diff
@@ -25,72 +25,122 @@ protected function getMiddleware()
         return new AuthHttpBasic(
             $this->authDriver,
             $this->registry
+            
         );
     }
    
 
-    public function testNotAuthenticatedWithoutHeader()  //test ob user nicht authentifiziert
+    public function testNotAuthenticatedWithoutHeader()  
     {
         $username = 'testUser01';
-        
-        $this->authDriver->method('authenticate')->willReturn(false); //not authenticated
+
+        //not authenticated
+        $this->authDriver->method('authenticate')->willReturn(false); 
+        //new login request
         $this->registry->method('getAuth')->willReturn($username);
+        
         $request = $this->requestFactory->createServerRequest('GET', '/test');
         $middleware = $this->getMiddleware();
         $response = $middleware->process($request, $this->handler);
         
-        $noAuthHeader = $this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); //ohne authorization header
-     
-        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER'); //nicht authorisiert->willreturn(false)
+        //no authorization header
+        $noAuthHeader = $this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); 
+        //NULL = notAuthenticated
+        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER'); 
         
-        $this->assertNull($notAuth); //not authenticated
+        $this->assertNull($notAuth); 
         $this->assertEquals($username, $noAuthHeader); 
        
         
       
     }
     
-    public function testNotAuthenticatedWithHeader()  //test ob user nicht authentifiziert
+    public function testNotAuthenticatedWithHeader()  
     {
         $username = 'testUser01';
         
+        //not authenticated
         $this->authDriver->method('authenticate')->willReturn(false);
+
         $this->registry->method('getAuth')->willReturn($username);
-        $request = $this->requestFactory->createServerRequest('GET', '/test');
+        
+        
+        //with authorization header 
+        $request = $this->requestFactory->createServerRequest('GET', '/test')->withHeader('Authorization','BASIC YXNkZjphc2QxMjM=');
         $middleware = $this->getMiddleware();
         $response = $middleware->process($request, $this->handler);
         
-        $AuthHeader = !$this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); //mit authorization header
-        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER');
+        $authHeader = $this->recentlyHandledRequest->hasHeader('Authorization');
 
-        $this->assertFalse($AuthHeader); //not no_auth_header
+        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER');
```

Comment:
I would rename this to `$authenticatedUser`.
Then `$this->assertNull($authenticatedUser);  ` becomes more readable